ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -0,0 +1,350 @@
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+  /* static methods */
+  static void init_filter_llama ();
+  static void fini_filter_llama ();
+
+  private:
+  static TensorFilterLlamaCpp *registered;
+  static const char *name;
+
+  llama_context *ctx;
+  llama_model *model;
+  llama_sampler *smpl;
+  llama_model_params model_params;
+  llama_sampler_chain_params sampler_params;
+  llama_context_params ctx_params;
+};
+
+void init_filter_llama (void) __attribute__ ((constructor));
+void fini_filter_llama (void) __attribute__ ((destructor));
+
+TensorFilterLlamaCpp *TensorFilterLlamaCpp::registered = nullptr;
+const char *TensorFilterLlamaCpp::name = "llamacpp";
+
+/**
+ * @brief Construct a new llamacpp subplugin instance
+ */
+TensorFilterLlamaCpp::TensorFilterLlamaCpp ()
+    : tensor_filter_subplugin (), ctx (nullptr), model (nullptr), smpl (nullptr)
+{
+}
+
+/**
+ * @brief Destructor of TensorFilterllamacpp
+ */
+TensorFilterLlamaCpp::~TensorFilterLlamaCpp ()
+{
+  if (model) {
+    llama_free_model (model);
+    model = nullptr;
+  }
+
+  if (smpl) {
+    llama_sampler_free (smpl);
+    smpl = nullptr;
+  }
+}
+
+/**
+ * @brief Method to get an empty object
+ */
+tensor_filter_subplugin &
+TensorFilterLlamaCpp::getEmptyInstance ()
+{
+  return *(new TensorFilterLlamaCpp ());
+}
+
+/**
+ * @brief Configure llamacpp instance
+ */
+void
+TensorFilterLlamaCpp::configure_instance (const GstTensorFilterProperties *prop)
+{
+  /** @todo: Get ngl value from prop */
+  int ngl = 99;
+
+  if (!prop->invoke_dynamic) {
+    throw std::invalid_argument (
+        "llamacpp only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
+  }
+
+  /** load dynamic backends */
+  ggml_backend_load_all ();
+  model_params = llama_model_default_params ();
+  model_params.n_gpu_layers = ngl;
+
+  model = llama_load_model_from_file ((char *) prop->model_files[0], model_params);
+  if (model == nullptr) {
+    throw std::invalid_argument ("Failed to load model");
+  }
+
+  ctx_params = llama_context_default_params ();
+
+  /** enable performance counters */
+  ctx_params.no_perf = false;
+
+  sampler_params = llama_sampler_chain_default_params ();
+  sampler_params.no_perf = false;
+  smpl = llama_sampler_chain_init (sampler_params);
+
+  llama_sampler_chain_add (smpl, llama_sampler_init_greedy ());
+}
+
+/**
+ * @brief Invoke llamacpp using input tensors
+ */
+void
+TensorFilterLlamaCpp::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  UNUSED (input);
+  UNUSED (output);
+
+  throw std::runtime_error (
+      "llamacpp only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
+}
+
+/**
+ * @brief Invoke llamacpp using input tensors
+ */
+void
+TensorFilterLlamaCpp::invoke_dynamic (GstTensorFilterProperties *prop,
+    const GstTensorMemory *input, GstTensorMemory *output)
+{
+  /** @todo: Get n_predict value from prop */
+  int n_prompt, n_pos, n_predict = 32, n_decode = 0;
```

Comment:
I'm wondering if ``` n_predict' '' is fixed or if it will be brought in as a property.

Reply:
It's not fixed so I will updated to use property later.
```  /** @todo: Get n_predict value from prop */```

Reply:
Thank you.